server/src/main/java/org/elasticsearch/index/translog/TruncateTranslogAction.java

```diff
@@ -209,7 +210,7 @@ private static void writeEmptyCheckpoint(Path filename, int translogLength, long
     private static int writeEmptyTranslog(Path filename, String translogUUID) throws IOException {
         try (FileChannel fc = FileChannel.open(filename, StandardOpenOption.WRITE, StandardOpenOption.CREATE_NEW)) {
             TranslogHeader header = new TranslogHeader(translogUUID, SequenceNumbers.UNASSIGNED_PRIMARY_TERM);
-            header.write(fc);
+            header.write(fc, true); // TODO: make fsync conditional on IndexModule#NODE_STORE_USE_FSYNC?
```

Comment:
NIT: some todos left, but I think it is okay to address them in a followup.
I do not think this action is used often

Reply:
Yea it's a rare outlier, I didn't feel like spending time on it yet, maybe later :)